dbms/src/Storages/TrivialBuffer.cpp

```diff
@@ -0,0 +1,554 @@
+#include <Interpreters/InterpreterSelectQuery.h>
+#include <Interpreters/InterpreterInsertQuery.h>
+#include <Interpreters/InterpreterAlterQuery.h>
+#include <DataStreams/IProfilingBlockInputStream.h>
+#include <Databases/IDatabase.h>
+#include <Storages/TrivialBuffer.h>
+#include <Parsers/ASTInsertQuery.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTExpressionList.h>
+#include <Common/setThreadName.h>
+#include <Common/CurrentMetrics.h>
+#include <Common/SipHash.h>
+#include <common/logger_useful.h>
+#include <Poco/Ext/ThreadNumber.h>
+
+#include <ext/range.hpp>
+
+
+namespace ProfileEvents
+{
+    extern const Event TrivialBufferFlush;
```

Comment:
No problem if you will share metrics and events name with Buffer.
BTW, this PR does not have corresponding variable definitions.